src/js/app.js

```diff
@@ -0,0 +1,164 @@
+    };
+
+    $scope.searchIndex = () => {
+      $scope.searchResult = {};
+      if (typeof $scope.bookIndices === 'undefined') {
+        $scope.msgDisplay('danger', 'No files has been indexed Yet');
+        return false;
+      }
+      const keywords = $scope.keyword,
+        ms = Object.keys($scope.selection)
+          .filter(key => $scope.selection[key] === true);
+      let totalBooks;
+      if (ms.length < 1) {
+        totalBooks = Object.keys($scope.indexed);
+      } else {
+        totalBooks = ms;
+      }
+      try {
+        invertedIndex.searchIndex(keywords, totalBooks);
+        const finalResult = invertedIndex.finalResult,
+          keys = Object.keys(finalResult);
+        keys.forEach((key) => {
+          const file = finalResult[key];
+          const totalBook = invertedIndex.booksIndex(key);
+          $scope.searchResult[key] = {
+            name: key,
+            result: file,
+            countBook: totalBook
+          };
+          if (Object.keys($scope.searchResult).length !== 0) {
+            $scope.showSearch = true;
+            $scope.showIndex = false;
+            $scope.welcome = false;
+          }
+        });
+      } catch (err) {
+        $scope.msgDisplay('danger', err);
+      }
+    };
+
+    $scope.home = () => {
+      $scope.showSearch = false;
+      $scope.showIndex = false;
+      $scope.welcome = true;
+    };
+
+    /** @function init
+     * @param {uploadedFile} uploadedFile
+   * @return {null} null
+  */
+    function uploadFile(uploadedFile) {
+      const files = uploadedFile.target.files;
+      $scope.replaceFiles = [];
+      const keys = Object.keys(files);
+      keys.forEach((key) => {
+        InvertedIndex.readFile(files[key])
+          .then((result) => {
+            const fileName = result[0];
+            const fileContent = result[1];
+            $scope.$apply(() => {
+              $scope.uploadedFiles[fileName] = fileContent;
+              const msg = `${fileName} was uploaded successfully`;
+              $.toaster({
+                priority: 'success',
+                title: 'SUCCESS!',
+                message: msg
+              });
+            });
+          })
+          .catch((error) => {
+            $scope.msgDisplay('danger', error);
+            $scope.$apply();
+          });
+      });
+    }
+    // We can attach the `fileselect` event to all file inputs on the page
+    // const fileUpload = document.getElementById('upload');
+    /** @function init
+   * @return {null} null
+  */
+    function Init() {
+      const fileselect = document.getElementById('fileselect');
+      fileselect.addEventListener('change', uploadFile, false);
+    }
+    if (window.File && window.FileList && window.FileReader) {
+      Init();
+    }
+  }
+  /** @function init
+   * @return {null} null
+  */
+  const table = () => (input, arr) => {
+    if (arr.includes(input)) {
+      return 'check';
+    }
+    return 'cross';
+  };
+  // };
```

Comment:
Same here (Dead Code), Also your JSDoc return statement is null, but you are returning string in your function block